src/javascript/note.js

```diff
@@ -13,9 +13,11 @@ export default class Note {
     this.isHoldNote = keyObj.h && keyObj.h[key];
     this.isUserHolding = false;
     this.isHoldingDone = false;
+    this.didUserHold = false;
 
-    this.y = 0;
+    this.y = y ?? 0;
     this.singleNoteHeight = 10;
+    if (y) this.createdNote = true;
```

Comment:
What is this checking? If `y === 0`? If so, specify that you are comparing against a `number`, instead of relying on coercion semantics.

```suggestion
    if (y === 0) this.createdNote = true;
```

Reply:
I think the logic here is that if Y is not provided, then that means this is a playable note instead of a creating note (e.g. user is still holding on this note), so this might cause some problem here to make it a default value... I won't add this to batch for now